src/averageAges.js

```diff
@@ -59,6 +91,29 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const mothers = people
+    .filter(mom => people
+      .some(child => child.mother === mom.name));
+
+  const childrenWithMother = people
+    .filter(child =>
+      (!onlyWithSon && mothers
+        .some(mom => mom.name === child.mother))
+    || (onlyWithSon && child.sex === SEX_MALE
+    && mothers
+      .some(mom => mom.name === child.mother))
+    );
```

Comment:
Now everything is written correctly, but such code is very difficult to read and maintain. Break it down into simpler parts.